File: ts-order-service/src/repositories/order-repository.ts
import { Order, OrderStatus } from '../models/order';

class OrderRepository {
  private orders: Map<string, Order> = new Map();

  findAll(): Order[] {
    return Array.from(this.orders.values());
  }

  findById(id: string): Order | undefined {
    return this.orders.get(id);
  }

  findByCustomerId(customerId: string): Order[] {
    return this.findAll().filter(order => order.customerId === customerId);
  }

  findByStatus(status: OrderStatus): Order[] {
    return this.findAll().filter(order => order.status === status);
  }

  save(order: Order): Order {
    this.orders.set(order.id, order);
    return order;
  }

  delete(id: string): boolean {
    return this.orders.delete(id);
  }
}

export default new OrderRepository();

File: order-kotlin-service/src/main/kotlin/de/hdi/order_kotlin_service/repository/OrderRepository.kt
package de.hdi.order_kotlin_service.repository

import de.hdi.order_kotlin_service.model.Order
import de.hdi.order_kotlin_service.model.OrderStatus
import org.springframework.stereotype.Repository
import java.util.concurrent.ConcurrentHashMap

@Repository
class OrderRepository {
    private val orders = ConcurrentHashMap<String, Order>()

    fun save(order: Order): Order {
        orders[order.id] = order
        return order
    }

    fun findById(id: String): Order? = orders[id]

    fun findAll(): List<Order> = orders.values.toList()

    fun update(id: String, status: OrderStatus): Order? {
        val order = orders[id] ?: return null
        val updatedOrder = order.copy(status = status)
        orders[id] = updatedOrder
        return updatedOrder
    }

    fun delete(id: String): Boolean = orders.remove(id) != null
}